### ISSUE TITLE
Integrations - Tuya - Devices - battery column sorted as string not as number

### ISSUE THREAD

**Issue by IonutDanNica** [home-assistant/frontend#13186]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When you try to sort for the battery level of devices under the tuya integration, the sorting is done as a string not as a percentage number. 
![image](https://user-images.githubusercontent.com/14056001/178692343-2388820c-4df7-4257-9177-6e670ec116d4.png)

### Describe the behavior you expected

List should be sorted by descending percentage value, not percentage string.

### Steps to reproduce the issue

1. Have any 2 battery operated devices in Tuya integration
2. Sort ascending/descending the devices tab
3.
...

### What version of Home Assistant Core has the issue?

Home Assistant Core 2022.7.3

### What was the last working version of Home Assistant Core?

Home Assistant Core 2022.7.3

### In which browser are you experiencing the issue with?

Chrome Version 103.0.5060.114 (Official Build) (64-bit)

### Which operating system are you using to run this browser?

Windows 10, 21H2

### State of relevant entities

```txt
{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2022.7.3",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.10.5",
    "docker": true,
    "arch": "x86_64",
    "timezone": "Europe/Bucharest",
    "os_name": "Linux",
    "os_version": "5.15.45",
    "supervisor": "2022.07.0",
    "host_os": "Home Assistant OS 8.2",
    "docker_version": "20.10.14",
    "chassis": "vm",
    "run_as_root": true
  },
  "custom_components": {
    "esxi_stats": {
      "version": "0.7.0",
      "requirements": [
        "pyvmomi==7.0.3"
      ]
    },
    "hacs": {
      "version": "1.25.5",
      "requirements": [
        "aiogithubapi>=22.2.4"
      ]
    }
  },
  "integration_manifest": {
    "domain": "tuya",
    "name": "Tuya",
    "documentation": "https://www.home-assistant.io/integrations/tuya",
    "requirements": [
      "tuya-iot-py-sdk==0.6.6"
    ],
    "dependencies": [
      "ffmpeg"
    ],
    "codeowners": [
      "@Tuya",
      "@zlinoliver",
      "@frenck"
    ],
    "config_flow": true,
    "iot_class": "cloud_push",
    "dhcp": [
      {
        "macaddress": "105A17*"
      },
      {
        "macaddress": "10D561*"
      },
      {
        "macaddress": "1869D8*"
      },
      {
        "macaddress": "381F8D*"
      },
      {
        "macaddress": "508A06*"
      },
      {
        "macaddress": "68572D*"
      },
      {
        "macaddress": "708976*"
      },
      {
        "macaddress": "7CF666*"
      },
      {
        "macaddress": "84E342*"
      },
      {
        "macaddress": "D4A651*"
      },
      {
        "macaddress": "D81F12*"
      }
    ],
    "loggers": [
      "tuya_iot"
    ],
    "is_built_in": true
  },
  "data": {
    "endpoint": "https://openapi.tuyaeu.com",
    "auth_type": 0,
    "country_code": "40",
    "app_type": "smartlife",
    "mqtt_connected": true,
    "disabled_by": null,
    "disabled_polling": false,
    "name": "W Olivia ",
    "model": "MC500A",
    "category": "mcs",
    "product_id": "2mbfxlzr",
    "product_name": "Contact Sensor",
    "online": true,
    "sub": true,
    "time_zone": "+03:00",
    "active_time": "2022-03-15T15:00:27+00:00",
    "create_time": "2022-03-15T15:00:27+00:00",
    "update_time": "2022-03-15T15:00:50+00:00",
    "function": {},
    "status_range": {
      "doorcontact_state": {
        "type": "Boolean",
        "value": {}
      },
      "battery_percentage": {
        "type": "Integer",
        "value": {
          "unit": "%",
          "min": 0,
          "max": 100,
          "scale": 0,
          "step": 1
        }
      }
    },
    "status": {
      "doorcontact_state": true,
      "battery_percentage": 84
    },
    "home_assistant": {
      "name": "W Olivia ",
      "name_by_user": null,
      "disabled": false,
      "disabled_by": null,
      "entities": [
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": "window",
          "original_device_class": "door",
          "icon": null,
          "original_icon": null,
          "unit_of_measurement": null,
          "state": {
            "entity_id": "binary_sensor.contact_sensor_2",
            "state": "on",
            "attributes": {
              "device_class": "window",
              "friendly_name": "W Olivia "
            },
            "last_changed": "2022-07-13T08:53:19.414581+00:00",
            "last_updated": "2022-07-13T08:53:19.414581+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "diagnostic",
          "device_class": null,
          "original_device_class": "battery",
          "icon": null,
          "original_icon": null,
          "unit_of_measurement": "%",
          "state": {
            "entity_id": "sensor.contact_sensor_2_battery",
            "state": "84.0",
            "attributes": {
              "state_class": "measurement",
              "unit_of_measurement": "%",
              "device_class": "battery",
              "friendly_name": "W Olivia  Battery"
            },
            "last_changed": "2022-07-13T08:53:19.443306+00:00",
            "last_updated": "2022-07-13T08:53:19.443306+00:00"
          }
        }
      ]
    }
  }
}



{
  "home_assistant": {
    "installation_type": "Home Assistant OS",
    "version": "2022.7.3",
    "dev": false,
    "hassio": true,
    "virtualenv": false,
    "python_version": "3.10.5",
    "docker": true,
    "arch": "x86_64",
    "timezone": "Europe/Bucharest",
    "os_name": "Linux",
    "os_version": "5.15.45",
    "supervisor": "2022.07.0",
    "host_os": "Home Assistant OS 8.2",
    "docker_version": "20.10.14",
    "chassis": "vm",
    "run_as_root": true
  },
  "custom_components": {
    "esxi_stats": {
      "version": "0.7.0",
      "requirements": [
        "pyvmomi==7.0.3"
      ]
    },
    "hacs": {
      "version": "1.25.5",
      "requirements": [
        "aiogithubapi>=22.2.4"
      ]
    }
  },
  "integration_manifest": {
    "domain": "tuya",
    "name": "Tuya",
    "documentation": "https://www.home-assistant.io/integrations/tuya",
    "requirements": [
      "tuya-iot-py-sdk==0.6.6"
    ],
    "dependencies": [
      "ffmpeg"
    ],
    "codeowners": [
      "@Tuya",
      "@zlinoliver",
      "@frenck"
    ],
    "config_flow": true,
    "iot_class": "cloud_push",
    "dhcp": [
      {
        "macaddress": "105A17*"
      },
      {
        "macaddress": "10D561*"
      },
      {
        "macaddress": "1869D8*"
      },
      {
        "macaddress": "381F8D*"
      },
      {
        "macaddress": "508A06*"
      },
      {
        "macaddress": "68572D*"
      },
      {
        "macaddress": "708976*"
      },
      {
        "macaddress": "7CF666*"
      },
      {
        "macaddress": "84E342*"
      },
      {
        "macaddress": "D4A651*"
      },
      {
        "macaddress": "D81F12*"
      }
    ],
    "loggers": [
      "tuya_iot"
    ],
    "is_built_in": true
  },
  "data": {
    "endpoint": "https://openapi.tuyaeu.com",
    "auth_type": 0,
    "country_code": "40",
    "app_type": "smartlife",
    "mqtt_connected": true,
    "disabled_by": null,
    "disabled_polling": false,
    "name": "W Kitchen",
    "model": "MC500A",
    "category": "mcs",
    "product_id": "2mbfxlzr",
    "product_name": "Contact Sensor",
    "online": true,
    "sub": true,
    "time_zone": "+03:00",
    "active_time": "2022-04-15T17:19:56+00:00",
    "create_time": "2022-04-15T17:19:56+00:00",
    "update_time": "2022-04-15T17:20:14+00:00",
    "function": {},
    "status_range": {
      "doorcontact_state": {
        "type": "Boolean",
        "value": {}
      },
      "battery_percentage": {
        "type": "Integer",
        "value": {
          "unit": "%",
          "min": 0,
          "max": 100,
          "scale": 0,
          "step": 1
        }
      }
    },
    "status": {
      "doorcontact_state": true,
      "battery_percentage": 9
    },
    "home_assistant": {
      "name": "W Kitchen",
      "name_by_user": null,
      "disabled": false,
      "disabled_by": null,
      "entities": [
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": null,
          "device_class": "window",
          "original_device_class": "door",
          "icon": null,
          "original_icon": null,
          "unit_of_measurement": null,
          "state": {
            "entity_id": "binary_sensor.contact_sensor_5",
            "state": "on",
            "attributes": {
              "device_class": "window",
              "friendly_name": "W Kitchen"
            },
            "last_changed": "2022-07-13T08:53:19.411372+00:00",
            "last_updated": "2022-07-13T08:53:19.411372+00:00"
          }
        },
        {
          "disabled": false,
          "disabled_by": null,
          "entity_category": "diagnostic",
          "device_class": null,
          "original_device_class": "battery",
          "icon": null,
          "original_icon": null,
          "unit_of_measurement": "%",
          "state": {
            "entity_id": "sensor.contact_sensor_5_battery",
            "state": "9.0",
            "attributes": {
              "state_class": "measurement",
              "unit_of_measurement": "%",
              "device_class": "battery",
              "friendly_name": "W Kitchen Battery"
            },
            "last_changed": "2022-07-13T08:53:19.441071+00:00",
            "last_updated": "2022-07-13T08:53:19.441071+00:00"
          }
        }
      ]
    }
  }
}
```

**Comment by Quentame** [comment]
I don't think it's Tuya related, but to sort function applied to this column.

Also, don't paste any of your `.[X].json` file, it may contain sensitive data, and here it doesn't seem relevant.

**Comment by Quentame** [comment]
Same issue with Z-Wave:
<img width="1381" alt="image" src="https://user-images.githubusercontent.com/14821482/219157264-cf317907-6e33-45f8-a8c3-a1c2afea0a55.png">

**Comment by karwosts** [comment]
@Quentame - what version are you on? The sorting of battery levels in devices column was fixed in 2023.2. See https://github.com/home-assistant/frontend/pull/14926

Though I'm not 100% sure if that table you are showing is the devices table, or some other table somewhere else that I haven't found.
